### ISSUE TITLE
Filter Queries on mongodb

### ISSUE THREAD

**Issue by Apoorva333** [compose/transporter#263]
## Feature Request

Hi,

I'm using transporter-0.1.2 to map data from mongodb to elasticsearch on RHEL 6. Is there any feature to perform filter queries on mongodb so that transporter can retrieve only the documents satisfying a particular condition ( like db.collection.find("flag" : true) ) rather than pulling out all the documents in the collection? 

PS: We are currently working on a large data set of around 50GB. So, we can save a lot of time if the transporter retrieves some specific documents from the collection.

Thank you

**Comment by jipperinbham** [comment]
@Apoorva333 I definitely think that's a feature worth investigating. Right now, our recommendation is to use transform functions to "skip" documents that you do not want sent down the pipeline to sources. 

However, we have a known issue in `0.1.2` with mongodb -> elasticsearch for this use case but has since been fixed in master [here](https://github.com/compose/transporter/pull/261). I plan on getting `0.1.3` out very soon and it would allow you to configure a transformer function like so:

`filter_by_flag.js`
```javascript
module.exports = function(msg) {
  if msg.data['flag'] { 
    return msg; 
  }
  return false;
}
```

and then the `app.js` would look something like:

```javascript
var pipeline = Source({name:"mongodb", namespace: "db.collection"})
  .transform({filename: "filter_by_flag.js", namespace: "db.collection"})
  .save({name:"elasticsearch", namespace: "db.collection"}) 
```

**Comment by Apoorva333** [comment]
Thanks for the quick reply.  We are currently using the transform functions itself to get relevant documents. As I said earlier, reading all the documents and then filtering is  time consuming. So, it would be better if we can do a filter query on mongo db to get the required documents and then map them to Elasticsearch using transporter.

**Comment by jipperinbham** [comment]
@Apoorva333 we just released `v0.1.3` as it contained a bug fix for skipping documents with a transformer function. 

re: 
> I think it would be better if you provide an option to specify the required filter query of mongodb in one of the configuration files and then you can implement it by doing a filter query on mongodb to get the required documents.

No argument there, I've added the proper labels to this issue so we can track it internally but PRs are always welcome. Right now, our focus is improving the overall reliability and stability of each adaptor as well as the core pipeline code.
